src/main/java/org/jenkinsci/plugins/envinject/service/EnvInjectMasterEnvVarsSetter.java

```diff
@@ -1,68 +1,26 @@
 package org.jenkinsci.plugins.envinject.service;
 
 import hudson.EnvVars;
-import hudson.Main;
-import hudson.Platform;
+import java.util.NavigableMap;
 import jenkins.security.MasterToSlaveCallable;
 import org.jenkinsci.lib.envinject.EnvInjectException;
 
-import java.lang.reflect.Field;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
-
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * @author Gregory Boissinot
  */
 public class EnvInjectMasterEnvVarsSetter extends MasterToSlaveCallable<Void, EnvInjectException> {
 
-    private @NonNull EnvVars enVars;
+    private @NonNull NavigableMap<String, String> enVars;
 
-    public EnvInjectMasterEnvVarsSetter(@NonNull EnvVars enVars) {
+    public EnvInjectMasterEnvVarsSetter(@NonNull NavigableMap<String, String> enVars) {
         this.enVars = enVars;
     }
 
-    private Field getModifiers() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
-        Method getDeclaredFields0 = Class.class.getDeclaredMethod("getDeclaredFields0", boolean.class);
-        getDeclaredFields0.setAccessible(true);
-        Field[] fields = (Field[]) getDeclaredFields0.invoke(Field.class, false);
-        Field modifiers = null;
-        for (Field each : fields) {
-            if ("modifiers".equals(each.getName())) {
-                modifiers = each;
-                break;
-            }
-        }
-        if (modifiers == null) {
-            throw new NoSuchFieldException();
-        }
-        return modifiers;
-    }
-
     @Override
-    public Void call() throws EnvInjectException {
-        try {
-            Field platformField = EnvVars.class.getDeclaredField("platform");
-            platformField.setAccessible(true);
-            platformField.set(enVars, Platform.current());
-            if (Main.isUnitTest || Main.isDevelopmentMode) {
-                enVars.remove("MAVEN_OPTS");
-            }
-            Field masterEnvVarsFiled = EnvVars.class.getDeclaredField("masterEnvVars");
-            masterEnvVarsFiled.setAccessible(true);
-            Field modifiersField = getModifiers();
-            modifiersField.setAccessible(true);
-            modifiersField.setInt(masterEnvVarsFiled, masterEnvVarsFiled.getModifiers() & ~Modifier.FINAL);
-            masterEnvVarsFiled.set(null, enVars);
-        } catch (IllegalAccessException | NoSuchFieldException iae) {
-            throw new EnvInjectException(iae);
-        } catch (InvocationTargetException | NoSuchMethodException e) {
-            throw new RuntimeException(e);
-        }
-
+    public Void call() {
+        EnvVars.masterEnvVars.putAll(enVars);
```

Comment:
Not sure how this would work with multiple calls to `EnvInjectMasterEnvVarsSetter#call`. This very well may be incorrect. :confused: